plugins/source/azure/resources/services/costmanagement/view_queries_fetch.go

```diff
@@ -30,6 +32,17 @@ func fetchViewQueries(ctx context.Context, meta schema.ClientMeta, parent *schem
 		return err
 	}
 
+	if (qd.Timeframe == nil || *qd.Timeframe != armcostmanagement.TimeframeTypeCustom) && qd.TimePeriod == nil && item.Properties.DateRange != nil {
+		qd.Timeframe, qd.TimePeriod = dateRangeToTimeFrame(time.Now().UTC(), *item.Properties.DateRange)
+		if qd.Timeframe != nil {
+			cl.Logger().Debug().Any("report_name", item.Name).Str("date_range", *item.Properties.DateRange).Any("resulting_time_period", qd.TimePeriod).Any("resulting_timeframe", qd.Timeframe).Msg("setting time period from date range")
+		}
+	}
+
+	if qd.Timeframe == nil && item.Properties.DateRange != nil {
+		return fmt.Errorf("could not convert date range %q to time period", *item.Properties.DateRange)
```

Comment:
If we don't know about this date range we simply refuse fetching incorrect data.